### DISCUSSION TITLE
Kafka external access with OCP routes

### DISCUSSION THREAD

**Discussion by britoaldan** [strimzi/strimzi-kafka-operator#7060] [Q&A]
if suppose my Advertised listener is configured as say  SSL://testkafkastrimzi-0.somdomain.com:32000
and a corresponding service and route is created for the same 
And By default the route is exposed to external world on port 443 

How does kafka producer work 
the initial connection  will be done as SSL://testkafkastrimzi-0.somdomain.com:443, this connection goes through, and  which inturn returns  the actual metadata which is the configure ADL for that broker 
SSL://testkafkastrimzi-0.somdomain.com:32000
when the client tries to establish a connection with SSL://testkafkastrimzi-0.somdomain.com:32000 
wont these connection fail ?

**Comment by scholzj** [comment]
When using `type: route` listener, why would you configure the advertised listener in Kafka to advertise port 32000?

**Comment by britoaldan** [comment]
hi @scholzj , 
If the advertised listener is not defined in that case the listeners will be used correct which is SSL://0.0.0.0:32000
and still we will have this issue

**Reply by scholzj** [reply]
Maybe you should start by sharing your Kafka CR with your configuration and some logs of things which you think don't work. But in most cases, all you need to do is to configure the listener with the right type and the operator configures everything on its own. You do not need to set any advertised hostnames. You just create the listener with `type: route` and the operator will create the OCP Routes and collect their addresses and configure the broker automatically with them.
